.github/workflows/aws-e2e-tests-non-root.yaml

```diff
@@ -0,0 +1,75 @@
+name: AWS E2E Tests (Non-root)
+run-name: AWS E2E Tests (Non-root) - ${{ github.run_id }} - @${{ github.actor }}
+
+on:
+  push:
+    branches:
+      - master
+      - branch/*
+  pull_request:
+    paths:
+      - '.github/workflows/aws-e2e-tests-non-root.yaml'
+      - '**.go'
+      - 'go.mod'
+      - 'go.sum'
+      - 'build.assets/Makefile'
+      - 'build.assets/Dockerfile*'
+      - 'Makefile'
+  merge_group:
+    paths:
+      - '.github/workflows/aws-e2e-tests-non-root.yaml'
+      - '**.go'
+      - 'go.mod'
+      - 'go.sum'
+      - 'build.assets/Makefile'
+      - 'build.assets/Dockerfile*'
+      - 'Makefile'
+
+env:
+  TEST_KUBE: true
+  AWS_REGION: us-west-2
+  GHA_ASSUME_ROLE: arn:aws:iam::307493967395:role/tf-aws-e2e-gha-role
+  KUBERNETES_SERVICE_ASSUME_ROLE: arn:aws:iam::307493967395:role/tf-eks-discovery-ci-cluster-kubernetes-service-access-role
+  DISCOVERY_SERVICE_ASSUME_ROLE: arn:aws:iam::307493967395:role/tf-eks-discovery-ci-cluster-discovery-service-access-role
+  DISCOVERED_CLUSTER_NAME: gha-discovery-ci
```

Comment:
do these need to be in the workflow file?

Reply:
asking because the RFD says:

> To achieve this, GitHub action will be allowed to assume a set of roles that will simulate the minimum required permissions that each Teleport service requires to perform the required actions. Teleport services will be configured to assume these roles when running the E2E tests - requires changes to the Teleport configuration - so we can test them in isolation.

For kube access, that's just EKS. For database access, we could either create a single "database discovery" role, or a role specific to each kind of database: RDS, Redshift, etc. 

I'd kind of prefer the latter, just to really make each service use the absolute minimum permissions necessary for tests. But I'd prefer to just make these constants in the test code instead of defining them in the workflow file.

Reply:
I specified them in the workflow because if you want to test locally using any of our `aws-dev` accounts, you will need to replace each entry. Because of that, it's better to configure them using env vars.

your case is a little more complex but I think you can use the same approach.